frozendict/frozendict.pyi

```diff
@@ -1,6 +1,57 @@
-try:
-    from ._frozendict import *
-except ImportError:
-    from .core import *
+import collections
+from _typeshed import Self
+from collections.abc import Iterable, Iterator, Mapping
+from typing import Any, Generic, NoReturn, Tuple, TypeVar, Union, overload
 
-FrozenOrderedDict = frozendict
+_KT = TypeVar("_KT")
+_VT = TypeVar("_VT")
+
+def frozendict_or(
+    self: Mapping[_KT, _VT], other: Mapping[_KT, _VT]
+) -> "frozendict[_KT, _VT]": ...
+
+class frozendict(Mapping[_KT, _VT], Generic[_KT, _VT]):
+    # Fake __init__ to describe what __new__ does:
+    dict_cls: type[dict[Any, Any]] = ...
+    @overload
+    def __init__(self, **kwargs: _VT) -> None: ...
+    @overload
+    def __init__(self, mapping: Mapping[_KT, _VT]) -> None: ...
+    @overload
+    def __init__(self, iterable: Iterable[tuple[_KT, _VT]]) -> None: ...
+
+    # Magic Methods:
+    def __getitem__(self, __key: _KT) -> _VT: ...
+    def __len__(self) -> int: ...
+    def __iter__(self) -> Iterator[_KT]: ...
+    def __hash__(self) -> int: ...
+    def __repr__(self) -> str: ...
+    def copy(self: Self) -> Self: ...
+    def __copy__(self: Self) -> Self: ...
+    def __deepcopy__(self: Self) -> Self: ...
+    # Omit __reduce__, its used for Pickle and we don't need the annotation in code.
```

Comment:
? Well, also __copy__ is used for copy library, etc. Why pickle is different?

Reply:
Pickle needs a reference to the class as well. If we leave it out, the only people affected are those developing pickle.

Reply:
I do not understand. What's the difference with `__copy__`?

Reply:
Anyway I think I'll merge, but have you tested it? There's a bad prototype here: https://github.com/Marco-Sulla/python-frozendict/blob/master/test/typed.py